microsoft/testsuites/gpu/gpusuite.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from lisa import Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.features import Gpu, SerialConsole
+from lisa.testsuite import simple_requirement
+from lisa.tools import Lsmod, Reboot
+from lisa.util import LisaException
+
+
+@TestSuiteMetadata(
+    area="gpu",
+    category="functional",
+    description="""
+    This test suite runs the gpu test cases.
+    """,
+)
+class gpu(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            This test case verfies kernel support for gpu.
+            TODO: validate the test params against GPU support test matrix
+                    when test matrix is available.
+        """,
+        requirement=simple_requirement(
+            supported_features=[Gpu],
+        ),
+    )
+    def validate_gpu_support(self, case_name: str, node: Node) -> None:
+        gpu_feature = node.features[Gpu]
+        gpu_feature.check_support()
+
+    @TestCaseMetadata(
+        description="""
+            This test case verfies if gpu drivers are loaded fine.
+
+            Steps:
+            1. Validate if the VM SKU is supported for GPU.
+            2. Install LIS drivers if not already installed for Fedora and its
+                derived distros. Reboot the node
+            3. Install required gpu drivers on the VM and reboot the node. Validate gpu
+                drivers can be loaded successfully.
+
+        """,
+        requirement=simple_requirement(
+            supported_features=[Gpu, SerialConsole],
+        ),
+    )
+    def validate_load_driver(self, case_name: str, node: Node) -> None:
```

Comment:
There is another runtime setting that make this case run on a new environment, so it can verify reboot and check panic. Please add a task to me, to specify this case need it on requirement level. I will add it.

Reply:
Here -
https://microsoft.visualstudio.com/OS/_workitems/edit/33665762